File: 240611/점수대 파악하기/find-out-the-score-range.py
scores = list(map(int, input().split()))

n_scores = []
for score in scores:
    if score == 0:
        break
    else:
        n_scores.append((score // 10) * 10)

scores_cntr = {score: 0 for score in range(10, 101, 10)}
for n_score in n_scores:
    if n_score in scores_cntr:
        scores_cntr[n_score] += 1

scores_sort = sorted([(n_score, cnts) for n_score, cnts in scores_cntr.items()], key = lambda x: x[0], reverse = True)
for n_score, cnts in scores_sort:
    print(f"{n_score} - {cnts}")

File: 251116/점수대 파악하기/find-out-the-score-range.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    vector<int> scores;
    int x;

    while (cin >> x) 
    {
        if (x == 0) break;
        scores.push_back(x);
    }

    vector<int> n_scores;
    for (int score : scores) 
    {
        n_scores.push_back((score / 10) * 10);
    }

    vector<int> scores_cntr(11, 0);  
    for (int ns : n_scores)
    {
        if (ns >= 10 && ns <= 100) 
        {
            scores_cntr[ns / 10]++;  
        }
    }

    for (int v = 100; v >= 10; v -= 10) 
    {
        cout << v << " - " << scores_cntr[v / 10] << '\n';
    }

    return 0;
}
